thinc/optimizers.py

```diff
@@ -354,4 +354,127 @@ def _adam(self, ops, weights, gradient, lr_scale, key, nr_upd):
         )
 
 
+class SWA(object):
+    """
+    After the optimization ran 'start_step' number of steps
+    it switches the learning-rate of the optimizer and starts
+    recording the moving averages of all weights.
+    """
+
+    def __init__(
+        self,
+        optimizer: Optimizer,
+        lr: FloatOrSeq,
+        *,
+        freq: int = 1,
+        start_step: Optional[int] = 0
+    ):
+        self.optimizer = optimizer
+        self.start_step = start_step
+        self.freq = freq
+        self.lr = lr
+        self.nr_update: Dict[Tuple[int, str], int] = defaultdict(int)
```

Comment:
Just an idea: I like `collections.Counter` for such purposes, since the name clearly states the semantics. It also has some useful additional methods (which you probably don't need here :)).

Reply:
haha I was thinking about the same thing that I would use a `Counter` to khm count. But yeah, I'm not using anything from it :D